src/PlusDataCollection/StealthLink/vtkPlusStealthLinkTracker.cxx

```diff
@@ -917,10 +987,36 @@ PlusStatus vtkPlusStealthLinkTracker::InternalUpdate()
         this->ToolTimeStampedUpdateWithoutFiltering(toolIterator->second->GetSourceId(), instrumentToTrackerTransform, TOOL_OUT_OF_VIEW, unfilteredTime, timeSystemSec);
       }
     }
+    // if the wanted Transform is any tool to Tracker that is not considered the current instrument
     else
     {
       vtkSmartPointer<vtkMatrix4x4> transformMatrixForNotTrackedTool = vtkSmartPointer<vtkMatrix4x4>::New();
-      this->ToolTimeStampedUpdateWithoutFiltering(toolIterator->second->GetSourceId(), transformMatrixForNotTrackedTool, TOOL_OUT_OF_VIEW, unfilteredTime, timeSystemSec);
+      // if we are tracking the current instrument at all times
+      if (STRCASECMP(toolIterator->second->GetCustomProperty("AlwaysTrack").c_str(), "TRUE") == 0)
+      {
+        MNavStealthLink::Instrument instrument;
+        if (!this->InternalShared->GetInstrumentData(instrument, toolIterator->second->GetPortName()))    //thread safe
+        {
+          return PLUS_FAIL;

```

Comment:
Does GetInstrumentData return bool or PlusStatus? In general, use PlusStatus to return success/fail.
Does GetInstrumentData log the error? If not, then log the error before returning.